frontend/lib/resolution/scope-queries.cpp

```diff
@@ -486,17 +505,49 @@ static bool doLookupInScope(Context* context,
     // Search parent scopes, if any, until a module is encountered
     const Scope* cur = nullptr;
     bool reachedModule = false;
+    bool skipClosestConditional = false;
+    bool didSkipConditional = false;
+
+    // This trickiness is required to implement correct scoping behavior
+    // for 'if-vars' in conditionals. The 'if-var' lives in the scope
+    // for the conditional, but it is not visible within the 'else'
+    // branch. Without this hack, we'd be able to see the 'if-var' in
+    // both branches. First, detect if the start scope is the else-block.
+    if (!scope->id().isEmpty())
+      if (auto ast = parsing::idToAst(context, scope->id()))
+        if (isElseBlockOfConditionalWithIfVar(context, ast))
+          skipClosestConditional = true;
+
     for (cur = scope->parentScope(); cur != nullptr; cur = cur->parentScope()) {
       if (asttags::isModule(cur->tag())) {
         reachedModule = true;
         break;
       }
 
+      auto ast = !cur->id().isEmpty() ? parsing::idToAst(context, cur->id())
+                                      : nullptr;
+
+      // We could be in a nested block, so check for the else-block to
+      // trigger the pattern matching as we walk up...
+      if (!skipClosestConditional && ast)
+        if (isElseBlockOfConditionalWithIfVar(context, ast))
+          skipClosestConditional = true;
+
+      // Skip the first conditional's scope if we need to.
+      if (skipClosestConditional && !didSkipConditional) {
+        if (ast && ast->isConditional()) {
+          didSkipConditional = true;
```

Comment:
I still don't see the need for both a `skipClosestConditional` and a `didSkipConditional` -- especially since you don't reset the latter? Does this code properly skip variables in nested else branches, like the following:

```Chapel
if var x = someClass {
} else if var y = someClass {
  writeln(x,y);
}
```

Reply:
Hmmmmm, yeah, I'll have to add another test(s) and I definitely don't think I handled something like that, thanks.